src/main/java/at/hannibal2/skyhanni/data/QuiverAPI.kt

```diff
@@ -0,0 +1,255 @@
+            storage?.arrows?.arrowAmount = value.map {
+                it.key.toString() to it.value
+            }.toMap().toMutableMap()
+        }
+    var currentAmount: Int
+        get() = arrowAmount[currentArrow]?.toInt() ?: 0
+        set(value) {
+            currentArrow?.let { arrow ->
+                arrowAmount.let { amount ->
+                    amount[arrow] = value.toFloat()
+                }
+            }
+        }
+
+    private var arrows: List<ArrowType> = listOf()
+
+    private val SKELETON_MASTER_CHESTPLATE = "SKELETON_MASTER_CHESTPLATE".asInternalName()
+
+    private val group = RepoPattern.group("data.quiver")
+    private val chatGroup = group.group("chat")
+    private val selectPattern by chatGroup.pattern("select", "§aYou set your selected arrow type to §f(?<arrow>.*)§a!")
+    private val fillUpJaxPattern by chatGroup.pattern(
+        "fillupjax",
+        "(§.)*Jax forged (§.)*(?<type>.*)(§.)* x(?<amount>.*)( (§.)*for (§.)*(?<coins>.*) Coins)?(§.)*!"
+    )
+    private val fillUpPattern by chatGroup.pattern(
+        "fillup",
+        "§aYou filled your quiver with §f(?<flintAmount>.*) §aextra arrows!"
+    )
+    private val clearedPattern by chatGroup.pattern("cleared", "§aCleared your quiver!")
+    private val arrowResetPattern by chatGroup.pattern("arrowreset", "§cYour favorite arrow has been reset!")
+    private val addedToQuiverPattern by chatGroup.pattern(
+        "addedtoquiver",
+        "(§.)*You've added (§.)*(?<type>.*) x(?<amount>.*) (§.)*to your quiver!"
+    )
+
+    // Bows that don't use the players arrows
+    private val fakeBowsPattern by group.pattern("fakebows", "^BOSS_SPIRIT_BOW$")
+    private val quiverInventoryNamePattern by group.pattern("quivername", "^Quiver$")
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (!isEnabled()) return
+        val message = event.message.trimWhiteSpace().removeResets()
+
+        selectPattern.matchMatcher(message) {
+            val arrow = group("arrow")
+            currentArrow = getArrowByNameOrNull(arrow) ?: return
+            currentAmount = arrowAmount[currentArrow]?.toInt() ?: 0
+
+            return
+        }
+
+        fillUpJaxPattern.matchMatcher(message) {
+            val type = group("type")
+            val amount = group("amount").formatNumber().toFloat()
+
+            val filledUpType = getArrowByNameOrNull(type) ?: return
+
+            val existingAmount = arrowAmount[filledUpType] ?: 0f
+            val newAmount = existingAmount + amount
+            arrowAmount[filledUpType] = newAmount
+
+            return
+        }
+
+        fillUpPattern.matchMatcher(message) {
+            val flintAmount = group("flintAmount").formatNumber().toFloat()
+            val existingAmount = arrowAmount[getArrowByNameOrNull("ARROW".asInternalName())] ?: 0f
+            val newAmount = existingAmount + flintAmount
+
+            arrowAmount[getArrowByNameOrFlint("ARROW".asInternalName())] = newAmount
```

Comment:
Same here, but also make ArrowType "ARROW" a constant variable, no need to look up the value all the time. Probably need a ```by Lazy``` to get it working.
```suggestion

            arrowAmount.merge(getArrowByNameOrFlint("ARROW".asInternalName()),flintAmount,Float::plus)
```